src/client/opamArg.ml

```diff
@@ -1525,8 +1528,14 @@ let pin ?(unpin_only=false) () =
     mk_flag ["e";"edit"] "With $(opam pin add), edit the opam file as with \
                           `opam pin edit' after pinning." in
   let kind =
-    let help = "Set the kind of pinning. Must be one of version, \
-                path, git, hg or darcs." in
+    let help =
+      "Sets the kind of pinning. Must be one of $(i,version), $(i,path), \
+       $(i,git), $(i,hg), $(i,darcs) or $(i,auto). \
```

Comment:
Since cmdliner 0.9.6 you can use [`Arg.doc_alts_enum`](http://erratique.ch/software/cmdliner/doc/Cmdliner.Arg.html#VALdoc_alts_enum) on `kinds` for rendering that kind of sentence.